src/challonge.service.js

```diff
@@ -1,121 +1,119 @@
-            R.map(({ participants }) => participants),
-            R.unnest,
-            R.map(p => [p.id, p]),
-            R.fromPairs,
-        )(tournamentsDetails);
+    const participantsById = R.pipe(
+        R.map(({ participants }) => participants),
+        R.unnest,
+        R.map(p => [p.id, p]),
+        R.fromPairs,
+    )(tournamentsDetails);
 
-        return R.pipe(
-            R.map(({ matches }) => matches),
-            R.unnest,
-            R.filter(m => ['pending', 'open'].includes(m.state)),
-            R.map(m => ({
-                ...m,
-                tournament: tournamentsById[m.tournament_id],
-                player1: m.player1_id ? participantsById[m.player1_id] : null,
-                player2: m.player2_id ? participantsById[m.player2_id] : null,
-            })),
-            R.filter(({ player1, player2 }) =>
-                player1 && player1.email_hash === memberEmailHash
-                    || player2 && player2.email_hash === memberEmailHash),
-        )(tournamentsDetails);
-    }
+    return R.pipe(
+        R.map(({ matches }) => matches),
+        R.unnest,
+        R.filter(m => ['pending', 'open'].includes(m.state)),
+        R.map(m => ({
+            ...m,
+            tournament: tournamentsById[m.tournament_id],
+            player1: m.player1_id ? participantsById[m.player1_id] : null,
+            player2: m.player2_id ? participantsById[m.player2_id] : null,
+        })),
+        R.filter(({ player1, player2 }) =>
+            player1 && player1.email_hash === memberEmailHash
+                || player2 && player2.email_hash === memberEmailHash),
+    )(tournamentsDetails);
+}
 
-    async function addTournamentParticipant(tournamentId, memberUsername, participantName = undefined) {
-        const { data } = await api.post(`tournaments/${tournamentId}/participants.json`, {
-            participant: {
-                challonge_username: memberUsername,
-                name: participantName,
-            },
-        });
-        return data.participant;
-    }
+const addTournamentParticipant = ({ api }) => async function (tournamentId, memberUsername, participantName = undefined) {
+    const { data } = await api.post(`tournaments/${tournamentId}/participants.json`, {
+        participant: {
+            challonge_username: memberUsername,
+            name: participantName,
+        },
+    });
+    return data.participant;
+}
 
-    return {
-        fetchAllTournaments,
-        fetchOpenTournaments,
-        fetchTournament,
-        fetchTournamentParticipants,
-        fetchMembers,
-        fetchOpenMatchesForMember,
-        addTournamentParticipant,
-    };
+const challongeService = {
+    fetchAllTournaments,
+    fetchOpenTournaments,
+    fetchTournament,
+    fetchTournamentParticipants,
+    fetchMembers,
+    fetchOpenMatchesForMember,
+    addTournamentParticipant,
 }
 
-module.exports = challongeServiceFactory;
+module.exports = challongeService;
```

Comment:
Is it possible to export this with the same pattern as the other modules, having the methods inline inside the `challongeService` declaration?

If not, could all other modules be changed to follow this pattern, with each method declared separately first?

Reply:
Note sure how fetchOpenMatchesForMember calls fetchOpenTournaments if they are inside the object.

Yes, changing the other modules can be done and would imo be nicer to read, because you remove one level of indentation.

Reply:
Sounds good. 👍 No need to do it in this PR (if you don't want to). :)